scripts/pysa_fuzzer/run2.py

```diff
@@ -0,0 +1,187 @@
+import subprocess
+import os
+import json
+import argparse
+from pathlib import Path
+import shutil
+import logging
+import random
+from code_generator2 import CodeGenerator
+
+logging.basicConfig(level=logging.INFO)
+
+def run_command(command, output_file=None):
+    try:
+        result = subprocess.run(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True, check=True)
+        if output_file:
+            with open(output_file, 'w') as file:
+                file.write(result.stdout)
+    except subprocess.CalledProcessError as e:
+        logging.error(f"Command '{command}' failed with error: {e.stderr}")
+
+def generate_random_seed(num_files, num_mutations=48):
+    return [random.randint(1, num_mutations) for _ in range(num_files)]
+
+def apply_mutation(generator, mutation_number):
+    if 1 <= mutation_number <= 24:
+        # Apply source mutation
+        mutation_method = getattr(generator, f"source_mutation_{mutation_number}")
+        mutation_method()
+    elif 25 <= mutation_number <= 48:
+        # Apply sink mutation
+        mutation_method = getattr(generator, f"sink_mutation_{mutation_number - 24}")
+        mutation_method()
+
+def generate_python_files(num_files, seed):
+    generator = CodeGenerator()
+    output_dir = Path('generated_files')
+
+    output_dir.mkdir(exist_ok=True)
+
+    filenames = []
+    mutation_mapping = {}
+    for i in range(num_files):
+        mutation_number = seed[i]
+        apply_mutation(generator, mutation_number)
+        generated_code = generator.generate()
+        filename = output_dir / f'test_{i+1}.py'
+        with open(filename, 'w') as file:
+            file.write("import random\n")
```

Comment:
this should be generated directly by `CodeGenerator.generate()` instead of being added here?